bastionzero/zli
# Distinguish targets by environment for `SSH`

## Description of the change

The purpose of this PR is to update `zli generate sshConfig` to now generate host names that specify the environment of the target to be able to distinguish between bzero targets that have the same name. 

For example:
If we have:
- An environment with `Name: sai-test-env`
- A target with `Name: some-bzero-target`

The config file, `dev-bzero-bz-config`, will include the following Host names:
`Host some-bzero-target some-bzero-target.sai-test-env`

We are excluding a host name that includes the environment ID and only including a host name option with environment name because environments are unique. This should declutter the config file a little as well. 

This will allow users to be able to connect via `ssh some-bzero-target.sai-test-env` and distinguish between targets with the same name using the environment name when connecting via ssh. 

This feature does not support SSM targets and will only be able to distinguish between Bzero targets that have the same name. So for any SSM target, the only host name will be the target name. 

## Testing

To test this PR:
1. Create two bzero targets with the same name in different environments
2. Create a target connect policy for the two bzero targets that gives you, the user, `SSH Tunnel` acess
3. Run `bin/zli-macos generate sshConfig`
4. Open the `dev-bzero-bz-config` file to see the host names listed as described above. There should be one listing for each bzero target. The names of both the targets will be the same, but the environment names will be different. 
5. Connect to the two targets using a host name that includes the environment to distinguish between the targets. 

I have this set up in [my dev](https://saic.bastionzero.com/) so feel free to login, and I'll make you an admin. You'll just need to add yourself as a user to the two target connect policies I created.

**backend branch:** distinguish-ssh-by-env
**bzero branch:** 

#### Ready to run system tests?


## Relevant release note information

Release Notes: 

## Related JIRA tickets

Relates to JIRA: CWC-2027

## Relevant HotFix if applicable

Relevant HotFix PR Number:

## Have you considered the security impacts?

Does this PR have any security impact?


If yes, please explain:
